itez/users/api/views.py

```diff
@@ -12,25 +13,33 @@
     DestroyModelMixin
 )
 from rest_framework.response import Response
-from rest_framework.viewsets import GenericViewSet
+from rest_framework.viewsets import GenericViewSet, ViewSet
 
+from rolepermissions.roles import RolesManager
 from .serializers import (
     UserSerializer,
     ChangePasswordSerializer,
-    GroupModelSerializer
 )
 
 User = get_user_model()
 
 
 class UserViewSet(RetrieveModelMixin, ListModelMixin, CreateModelMixin, UpdateModelMixin, GenericViewSet):
+    permission_classes = [IsAuthenticated]
     serializer_class = UserSerializer
     queryset = User.objects.all()
     lookup_field = "username"
 
-    def get_queryset(self, *args, **kwargs):
-        assert isinstance(self.request.user.id, int)
-        return self.queryset.filter(id=self.request.user.id)
+    def list(self, request):
+        queryset = User.objects.all()
+        serializer = UserSerializer(queryset, many=True)
+        return Response(serializer.data)
+
+    def retrieve(self, request, username=None):
+        queryset = User.objects.all()
+        user = get_object_or_404(queryset, username=username)
```

Comment:
same  replacement here

Reply:
`user = get_object_or_404(queryset, username=username)`
`user = get_object_or_404(User, pk=user_id)`
OR
`user = User.objects.get(pk=user_id)`

Reply:
Refactored to query with `id`